MotionPhotoReader/app/src/main/java/com/google/android/libraries/motionphotoreader/TextureRender.java

```diff
@@ -82,21 +87,14 @@
             "  FragColor = texture(uTexUnit, TexCoord);\n" +
             "}";
 
-    // A single plane comprised of two triangles to hold the image texture
-    private final float[] triangleVerticesData = {
-            // positions (x,y)
-            -1.0f, -1.0f,   // bottom left
-             1.0f, -1.0f,   // bottom right
-            -1.0f,  1.0f,   // top left
-             1.0f,  1.0f    // top right
-    };
-
     private float[] uMatrix = new float[16];
+    private float[] uStabMatrix = new float[16];
```

Comment:
these two can probably be made `final` . Note that even though they are final, the contents can still be mutated.